lib/hooks/useSiteAggregation.ts

```diff
@@ -0,0 +1,93 @@
+import useSWR from 'swr';
+import { siteListFetcher, manyForecastDataFetcher } from './utils';
+import { ForecastDataPoint } from '../types';
+
+/**
+ * Sums the capacity and forecasts of multiple solar sites across
+ * all dates reported by the pv-sites API.
+ * @param allSiteUUID A list of UUIDs corresponding to multiple solar sites
+ * @return Aggregated predictions sorted by datetime
+ */
+const useSiteAggregation = (allSiteUUID: string[]) => {
+  const {
+    data: siteListData,
+    error: siteListError,
+    isLoading: isSiteListLoading,
+  } = useSWR(
+    `${process.env.NEXT_PUBLIC_API_BASE_URL}/api/sites`,
+    siteListFetcher
+  );
+
+  const parsedAllSiteUUID = allSiteUUID
+    .reduce((prev, curr) => `${prev},${curr}`)
+    .substring(1);
+
+  const {
+    data: manyForecastData,
+    error: manyForecastError,
+    isLoading: isManyForecastLoading,
+  } = useSWR(
+    `${process.env.NEXT_PUBLIC_API_BASE_URL}/api/pv_forecast?site_uuids=${parsedAllSiteUUID}`,
+    manyForecastDataFetcher
+  );
+
+  const error = AggregateError([manyForecastError, siteListError]);
+  const isLoading = isSiteListLoading || isManyForecastLoading;
+
+  // Sum the installed capacity at all of the sites
+  const totalInstalledCapacityKw = siteListData
+    ? siteListData.site_list.reduce(
+        (prevTotalSiteCapacity, newSite) =>
+          prevTotalSiteCapacity + newSite.installed_capacity_kw,
+        0
+      )
+    : 0;
```

Comment:
maybe we should be treating this as `undefined` rather than 0 when siteListData is falsy?